src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,80 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    private void invokeAdd(Method m, Object[] args) throws ExecutionException {
+        if (params.isImmutable()) {
+            throw new IllegalStateException("Could not call set on immutable object");
+        }
+        String attrName = Character.toLowerCase(m.getName().charAt(5)) + m.getName().substring(6);
+        AttributeInfo attributeInfo = typeInfoCache.get(clazz).get(attrName);
+        Object values = internal.get(getKeyName(clazz, attrName));
+
+        ArrayList<Object> valuesArray = (values instanceof Collection<?>)
```

Comment:
Left hand side of declaration should always be an interface. E.g. List.

Reply:
[49a63fa](https://github.com/BlackBeltTechnology/structured-map-proxy/pull/82/commits/49a63faed3d96604466f669b0e028e0f30578d8b)